builtin/nomad/jobspec/releaser.go

```diff
@@ -0,0 +1,577 @@
+			groupHealthy := false
+			for !groupHealthy {
+				currentTaskGroupState = deploy.TaskGroups[group]
+				if currentTaskGroupState.HealthyAllocs < len(currentTaskGroupState.PlacedCanaries) {
+					select {
+					case <-ticker.C:
+					case <-ctx.Done(): // cancelled
+						return status.Errorf(codes.Aborted, "Healthy deadline reached: %s", ctx.Err())
+					}
+					deploy, _, err = jobClient.LatestDeployment(*job.ID, q)
+					currentTaskGroupState = deploy.TaskGroups[group]
+					log.Info(fmt.Sprintf("Task group not healthy: %s", group))
+				} else {
+					groupHealthy = true
+				}
+			}
+		}
+	}
+
+	wq := &api.WriteOptions{
+		Namespace: *job.Namespace,
+	}
+
+	var u *api.DeploymentUpdateResponse
+	if r.config.FailDeployment {
+		u, _, err = deploymentClient.Fail(deploy.ID, wq)
+	} else {
+		u, _, err = deploymentClient.PromoteGroups(deploy.ID, groupsToPromote, wq)
+	}
+	if err != nil {
+		return err
+	}
+
+	st.Update(fmt.Sprintf("Monitoring evaluation %s", string(u.EvalID)))
+	if err := nomad.NewMonitor(st, client.NomadClient).Monitor(u.EvalID); err != nil {
+		return err
+	}
+
+	// TODO: Automatically search for Consul service, determine FQDN for service
+	// TODO: Automatically search for ingress gateway, determine FQDN for service
+	// TODO: Automatically search for IP and port of random Nomad alloc in job
+	// If meta not set, URL is empty
+	result.Url = job.Meta["waypoint.hashicorp.com/release_url"]
+	return nil
+}
+
+func (r *Releaser) resourceJobDestroy(
+	client *nomadClient,
+	state *Resource_Job,
+	sg terminal.StepGroup,
+) error {
+	// Do nothing because the platform will destroy the job for us
+	return nil
+}
+
+func (r *Releaser) resourceJobStatus(
+	ctx context.Context,
+	log hclog.Logger,
+	sg terminal.StepGroup,
+	state *Resource_Job,
+	client *nomadClient,
+	sr *resource.StatusResponse,
+) error {
+	s := sg.Add("Checking status of Nomad job resource %q...", state.Name)
+	defer s.Abort()
+
+	jobClient := client.NomadClient.Jobs()
+	s.Update("Getting job...")
+	// TODO: Because we don't have the namespace from the jobspec, we rely on the
+	//   NOMAD_NAMESPACE env var/searching for job via prefix- consider passing namespace
+	//   from deploy phase
+	jobs, _, err := jobClient.PrefixList(state.Name)
+	if err != nil {
+		return err
+	}
+
+	jobResource := sdk.StatusReport_Resource{
+		CategoryDisplayHint: sdk.ResourceCategoryDisplayHint_INSTANCE_MANAGER,
+	}
+	sr.Resources = append(sr.Resources, &jobResource)
+
+	s.Update("Getting job info...")
+	q := &api.QueryOptions{Namespace: jobs[0].JobSummary.Namespace}
+	job, _, err := jobClient.Info(jobs[0].ID, q)
+
+	if err != nil {
+		s.Update("No job was found")
+		s.Status(terminal.StatusError)
+		s.Done()
+		s = sg.Add("")
+
+		jobResource.Name = state.Name
+		jobResource.Health = sdk.StatusReport_MISSING
+		jobResource.HealthMessage = sdk.StatusReport_MISSING.String()
+	} else if job == nil {
+		return status.Errorf(codes.FailedPrecondition, "Nomad job response cannot be empty")
```

Comment:
Instead of failing the entire status job, we could make this case an UNKNOWN health with this as the health message 🤔